### PR TITLE
Remove orphaned refs, update reftools, switch to yaml, add oas3 example

### PR DESCRIPTION
**Pull request by bzmw**
This PR does a few things!
 - Update reftools to 1.0.8
 - Switch from js-yaml to yaml
 - Add an oas3 example

Also I fixed an issue I was having with orphaned references.
This issue was that parameters that were being filtered were not having their references removes from other schemas or paths. 
To fix it I record all filtered paths, and then recurse again and remove them.

I also included a version bump cause this is a minor change.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 'use strict';
 
 const recurse = require('reftools/lib/recurse.js').recurse;
-const clone = require('reftools/lib/clone.js').clone;
+const clone = require('reftools/lib/clone.js').deepClone;
```
Comment (MikeRalphson) [inline index.js]: Why the need to change the clone strategy?

File: index.js [hunk 2]
```diff
@@ -46,6 +47,11 @@ function filter(obj,options) {
             });
         }
     });
+    recurse(src,{},function(obj,key,state){
+        if (obj.hasOwnProperty('$ref') && filteredpaths.includes(obj['$ref'])) {
+            state.parent.splice(state.pkey, 1);
```
Comment (MikeRalphson) [inline index.js]: We should at least have a check here that state.parent is an Array.

File: openapi-filter.js [hunk 3]
```diff
@@ -30,20 +31,22 @@ let argv = require('yargs')
     .number('lineWidth')
     .alias('l', 'lineWidth')
     .describe('lineWidth','max line width of yaml output')
-    .default('lineWidth',-1)
+    .default('lineWidth',80)
```
Comment (MikeRalphson) [inline openapi-filter.js]: Making the max line width 80 here could be a breaking change for people. Can we not?

File: openapi-filter.js [hunk 4]
```diff
@@ -30,20 +31,22 @@ let argv = require('yargs')
...
+    .alias('i','inverse')
```
Comment (MikeRalphson) [inline openapi-filter.js]: You seem to have reinstated this duplicate line when a merge went wrong.  Can you remove?

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "openapi-filter",
-  "version": "1.3.0",
+  "version": "1.4.0",
```
Comment (MikeRalphson) [inline package.json]: Please don't bump versions in PRs, that's the job of a release step.

File: index.js (hunk 1 above)
Comment (bzmw) [reply index.js]: Ah sorry didn't mean to commit this line. Revert!

**Comment by bzmw** [comment]
@MikeRalphson thanks for the review, I've addressed all comments could you take another look please

**Comment by MikeRalphson** [comment]
Looking good now, thanks.

**Pull request merged**